client/app/cloud/project/billing/cloud-project-billing.service.js

```diff
@@ -1,227 +1,49 @@
-        self.data.hourlyBilling.hourlyUsage.volume,
-        (sum, volume) => sum + roundNumber(volume.totalPrice, 2),
-        0,
-      );
-      self.data.totals.hourly.volume = roundNumber(self.data.totals.hourly.volume, 2);
-    }
-
-    function initInstanceBandwidth() {
-      if (!self.data.hourlyBilling || !self.data.hourlyBilling.hourlyUsage) {
-        return;
-      }
-      const bandwidthByRegions = _.map(
-        self.data.hourlyBilling.hourlyUsage.instanceBandwidth,
-        (bandwidthByRegion) => {
-          const newBandwidthByRegion = _.clone(bandwidthByRegion);
-          newBandwidthByRegion.outgoingBandwidth.totalPrice = roundNumber(
-            newBandwidthByRegion.outgoingBandwidth.totalPrice,
-            2,
-          );
-          if (newBandwidthByRegion.outgoingBandwidth.quantity.value > 0) {
-            newBandwidthByRegion.outgoingBandwidth.quantity.value = roundNumber(
-              newBandwidthByRegion.outgoingBandwidth.quantity.value,
-              2,
-            );
-          }
-          return newBandwidthByRegion;
-        },
-      );
-      self.data.bandwidthByRegions = bandwidthByRegions;
-      self.data.totals.hourly.bandwidth = _.reduce(
-        self.data.bandwidthByRegions,
-        (sum, bandwidth) => sum + bandwidth.outgoingBandwidth.totalPrice,
-        0,
-      );
-      self.data.totals.hourly.bandwidth = roundNumber(self.data.totals.hourly.bandwidth, 2);
-    }
-
-    self.getConsumptionDetails = function (hourlyBillingInfo, monthlyBillingInfo) {
-      return self.getDataInitialized()
-        .then(() => {
-          self.data.hourlyBilling = hourlyBillingInfo;
-          self.data.monthlyBilling = monthlyBillingInfo;
-
-          return $q
-            .allSettled([
-              initHourlyInstanceList(),
-              initMonthlyInstanceList(),
-              initObjectStorageList(),
-              initArchiveStorageList(),
-              initSnapshotList(),
-              initVolumeList(),
-              initInstanceBandwidth(),
-            ])
-            .then(() => {
-              self.data.totals.monthly.total = roundNumber(self.data.totals.monthly.instance, 2);
-              self.data.totals.hourly.total = roundNumber(
-                self.data.totals.hourly.instance
-                + self.data.totals.hourly.snapshot
-                + self.data.totals.hourly.objectStorage
-                + self.data.totals.hourly.archiveStorage
-                + self.data.totals.hourly.volume
-                + self.data.totals.hourly.bandwidth,
-                2,
-              );
-              self.data.totals.total = roundNumber(
-                self.data.totals.monthly.total + self.data.totals.hourly.total,
-                2,
-              );
-              return self.data;
-            });
+      static formatPrice(value, currencyCode) {
+        return ({
+          value,
+          text: `${value.toFixed(2)} ${currencyCode}`,
```

Comment:
This doesn't respect internationalization :)

For example, Americans use this notation: `$100` :)

You can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat :)

Reply:
It is to keep the same format than is given by the API and so I don't have to format everything